### DISCUSSION TITLE
Error while trying to clone and install the foundry-zksync

### DISCUSSION THREAD

**Discussion by Oyedare** [Cyfrin/foundry-full-course-cu#2440] [Q&A]
Anytime i run the command for installing the foundry zk-sync on git bash, i run into an error

Error:
```
Archive:  /tmp/tmp.5ZX87catGH/foundry-zksync.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /tmp/tmp.5ZX87catGH/foundry-zksync.zip or
        /tmp/tmp.5ZX87catGH/foundry-zksync.zip.zip, and cannot find /tmp/tmp.5ZX87catGH/foundry-zksync.zip.ZIP, period.
foundryup-zksync: command failed: unzip /tmp/tmp.5ZX87catGH/foundry-zksync.zip -d /c/Users/HP/.foundry/bin

```
What might have caused it, is it that the cloning process wasn't complete? Though i made sure it cloned successfully. Or might there be another thing causing that error

**Comment by EngrPips** [comment]
Hello @Oyedare, Please try with this `https://github.com/matter-labs/foundry-zksync` let's see if it works.

**Reply by Oyedare** [reply]
Do you mean i should clone with that url? @EngrPips

**Reply by EngrPips** [reply]
Yes

**Reply by Oyedare** [reply]
I'm getting the same error. I would attach a screenshot of my CLI for you to have more info on the error
![Screenshot (170)](https://github.com/user-attachments/assets/a7bc3356-592a-4a2c-9646-7caa48ce333a)

**Reply by EngrPips** [reply]
Hello, sir @cromewar. Any help will be appreciated.

**Reply by gorgeoussarah** [reply]
I'm having the same issue and still haven't found any solution

**Reply by EngrPips** [reply]
Hello @PatrickAlphaC, We will appreciate your help here, sir.

**Reply by mumernisar** [reply]
@Oyedare @gorgeoussarah  From the picture in his comment I see that he is installing from 
```bash
./install-foundry-zksync
```
you have to go in one more folder into /foundryup-zksync.

I have installed zksync myself and that's not the install script I ran.!
you have to run them in the following order.
```bash 
git clone https://github.com/matter-labs/foundry-zksync.git
cd foundry-zksync/foundryup-zksync
./install
```

Hope that helps.
Note: I am using wsl on windows.! I dont know if it will work on git bash or not.!

**Reply by gorgeoussarah** [reply]
I'm still encountering the same error when trying to install the foundryup-zksync. I'm using Gitbash, but thanks for trying to help.

**Reply by mumernisar** [reply]
@gorgeoussarah Switch to wsl . It's very useful and quite easy to setup .
You can find a video on how to set it up on windows in the foundry basics section on cyfrin's website. If you can't find it lemme know I'll link it for you

**Reply by aniket892** [reply]
thanks a lot, i was really getting hard time with this .

**Reply by GoudhamT** [reply]
I got same error, switched to WSL and installed as stated above, it helped, thank you so much

**Comment by talhazullfiqar** [comment]
did you find any solution?

**Reply by mumernisar** [reply]
If nothing works try configuring wsl.

**Reply by talhazullfiqar** [reply]
hahhh thanksss man it worked 😭❤️

**Reply by mumernisar** [reply]
😂 np man ♥️

**Comment by KMean** [comment]
I had problems with wsl running the ./install-foundry-zksync from within the foundry-zksync folder as per:

https://github.com/zkSync-Community-Hub/zksync-developers/discussions/696

```
~/Projects/blockchain/foundry-zksync# ./install-foundry-zksync 
Downloading the install script...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2265  100  2265    0     0   8118      0 --:--:-- --:--:-- --:--:--  8118
Making install script executable...
Running the installation script...
Installing foundryup-zksync...

Detected your preferred shell is bash and added foundryup-zksync to PATH.
Run 'source /root/.bashrc' or start a new terminal session to use foundryup-zksync.
Then, simply run 'foundryup-zksync' to install foundry-zksync.
Sourcing the shell configuration file: '/root/.bashrc'
Downloading foundryup-zksync...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to create the file foundryup-zksync: Is a directory
  8  9945    8   823    0     0   6282      0  0:00:01 --:--:--  0:00:01  6282
curl: (23) Failed writing body (0 != 823)

```

running the script from the parent directory fixed the problem. Leave it here just in case someone is facing the same issue.

**Reply by EngrPips** [reply]
Awesome, I am glad you figured it out.

**Reply by Ukrys** [reply]
Thx!
